api/installation_service.go

```diff
@@ -0,0 +1,53 @@
+package api
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+)
+
+type InstallationService struct {
+	client   httpClient
+	progress progress
+}
+
+func NewInstallationService(client httpClient, progress progress) InstallationService {
+	return InstallationService{
+		client:   client,
+		progress: progress,
+	}
+}
+
+func (is InstallationService) Export(outputFile string) error {
```

Comment:
There was some discussion around the pattern for each one of these services today. The currently established pattern is to take in a struct for the input and return a struct + error for the output (even if ignored).

I see a couple of benefits to this but I welcome being challenged on this:
- Services have symmetry (all of them take an input and produce an output) this makes the API "guessable"
- If we ever decide to add / remove input or output arguments we end up only needing to touch the struct (and the implementation feels future proof without adding a higher support cost).

Does that make sense?